vars/pmmVersion.groovy

```diff
@@ -52,20 +51,32 @@ def call(String type='latest') {
     '2.38.1': 'ami-0c8a2742c5fef0023',
     '2.39.0': 'ami-079ca34c1b72b8e41',
   ]
-
-  List<String> versionsList = new ArrayList<>(versions.keySet());
+  List<String> versionsList = amiVersions.keySet() as List<String>;
   // Grab 5 latest versions
-  List<String> ovfVersions = ['2.39.0', '2.38.1', '2.38.0', '2.37.1', '2.37.0', '2.36.0'];
-  List<String> dbaasVersions = ['2.38.1', '2.38.0', '2.37.1', '2.37.0', '2.36.0', '2.35.0'];
+  List<String> ovfVersions = versionsList[-5..-1]
+  List<String> dbaasVersions = versionsList[-5..-1]
 
   switch(type) {
-    case 'latest':
-      def latestVersion = httpRequest "https://raw.githubusercontent.com/Percona-Lab/pmm-submodules/PMM-2.0/VERSION"
-      return latestVersion.content
+    case 'dev-latest':
+      sh(script: "sudo yum install -y wget jq")
+      String rcLatest = sh(
+              script: """wget -q "https://registry.hub.docker.com/v2/repositories/perconalab/pmm-client/tags?page_size=25&name=rc" -O - | jq -r .results[].name | grep '.*.*-rc\$' | sort -V | tail -n1""",
+              returnStdout: true
+      ).trim()
+      int major = rcLatest.split('\\.')[0] as Integer
+      int minor = rcLatest.split('\\.')[1] as Integer
+      return major + "." + ++minor + ".0"
+    case 'rc':
+      sh(script: "sudo yum install -y wget jq")
+      String rcLatest = sh(
+              script: """wget -q "https://registry.hub.docker.com/v2/repositories/perconalab/pmm-client/tags?page_size=25&name=rc" -O - | jq -r .results[].name | grep '.*.*-rc\$' | sort -V | tail -n1""",
```

Comment:
why do you check it with pmm-client?

Reply:
because pmm-client image is released in pair with server but it has no [el7/el9](https://hub.docker.com/layers/percona/pmm-server/2.40.0-el7/images/sha256-cbd835477c6b1ce5c1053c47218dd6740dca8c6d87c28a70a584a74a68180dd2) suffixes which needs to be parsed off. So the regex is just simpler.